### PR TITLE
Reverted tolerance for connectivity failure cases

### PR DESCRIPTION
**Pull request by brotherneeru**
This PR serves to turn Connectivity direct method tests to green. It removes the following tolerance for direct method tests

Ticket with details:https://msazure.visualstudio.com/One/_queries/edit/11005697/?action=new&triage=true

Before merging I will confirm that both connectivity and nested connectivity pipelines are passing.

### CODE REVIEW

File: test/modules/TestResultCoordinator/Reports/DirectMethod/Connectivity/DirectMethodConnectivityReportGenerator.cs [hunk 1]
```diff
@@ -162,24 +157,11 @@ public async Task<ITestResultReport> CreateReportAsync()
                     $"{dmSenderTestResult.GetFormattedResult()}. ReceiverTestResult: {dmReceiverTestResult.GetFormattedResult()}");
             }
 
-            bool didFindMatch = false;
             if (dmSenderTestResult.SequenceNumber == dmReceiverTestResult.SequenceNumber)
             {
-                dmReceiverTestResult = JsonConvert.DeserializeObject<DirectMethodTestResult>(receiverTestResults.Current.Result);
```
Comment (huguesBouvier) [inline test/modules/TestResultCoordinator/Reports/DirectMethod/Connectivity/DirectMethodConnectivityReportGenerator.cs:168]: I am not familiar with the code but what does that part do and why did you remove it?

Comment (brotherneeru) [reply test/modules/TestResultCoordinator/Reports/DirectMethod/Connectivity/DirectMethodConnectivityReportGenerator.cs:168]: This was just some code added for the earlier tolerances implemented by @and-rewsmith . Since we don't need it anymore on direct method tests, I removed it. Original PR is here: https://github.com/Azure/iotedge/pull/5512/files

File: e2e_deployment_files/connectivity_deployment.template.json [hunk 2]
```diff
@@ -897,20 +897,20 @@
             "TestReportType": "DirectMethodConnectivityReport",
             "SenderSource": "directMethodSender1.send",
             "ReceiverSource": "directMethodReceiver1.receive",
-            "TolerancePeriod": "00:01:00",
+            "TolerancePeriod": "00:03:00",
```
Comment (varunpuranik) [inline e2e_deployment_files/connectivity_deployment.template.json:900]: What does the tolerance period do? And why are we increasing it?

File: test/modules/TestResultCoordinator/Topology.cs [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-    {
-        // A single edge device.
-        SingleNode,
-
-        // Multiple edge devices connected together in some toplogy.
-        // Runs an extended test suite relative to `SingleNode`.
-        Nested
-    }
-}
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/Topology.cs:1]: revert?

**Comment by varunpuranik** [comment]
Is this PR Draft? If not, can you pls remove it from the title?

File: test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodConnectivityReportDataWithSenderAndReceiverSource.cs [hunk 4]
```diff
@@ -108,7 +90,7 @@ class DirectMethodConnectivityReportDataWithSenderAndReceiverSource
                     // NetworkOnFailure test
                     Enumerable.Range(1, 7).Select(v => (ulong)v),
                     new[] { 1UL, 2UL, 3UL, 5UL, 6UL, 7UL },
-                    new List<HttpStatusCode> { HttpStatusCode.OK, HttpStatusCode.OK, HttpStatusCode.InternalServerError, HttpStatusCode.NotFound, HttpStatusCode.InternalServerError, HttpStatusCode.InternalServerError, HttpStatusCode.InternalServerError },
+                    new List<HttpStatusCode> { HttpStatusCode.InternalServerError, HttpStatusCode.OK, HttpStatusCode.OK, HttpStatusCode.NotFound, HttpStatusCode.OK, HttpStatusCode.OK, HttpStatusCode.OK },
```
[the remark this answers is not in the document]
Comment (brotherneeru) [reply test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodConnectivityReportDataWithSenderAndReceiverSource.cs:93]: These are just all changes that were made when Andrew originally merged the tolerances in. I reverted all that had to do with directmethod tests and waiting for some comments from him

**Review by and-rewsmith — COMMENTED** [review]
Left a few nit comments. Have a meeting with Bilal to discuss other high level questions.

**Comment by and-rewsmith** [comment]
Discussed with @brotherneeru. We decided on the following steps:
1. There is one failure in single node connectivity run where a direct method came in 12 mins late. This doesn't make any sense as it exceeds the retry logic in SDKs and probably iothub. So we will have to investigate 1) any potential SDK ServiceClient exception returned when calling the direct method, 2) Kusto logs for hub to see if the direct method was handled properly by hub, 3) any EH logs that can further illuminate what happened.

   In the meantime, with this PR we should take the 30% failure tolerance and decrease it to something like 1% or 1 result so that the tests don't fail.

2. Nested connectivity pipeline is still running with the broker. We want to transition this to run without the broker and the results aren't meaningful, so the existing nested tolerances aren't needed. We should remove the nested tolerance for direct methods. Then other nested tolerance can be removed in other PRs (handling twins, messaging, etc.) along with the `Topology.cs` abstraction.
3. Add a comment or make variable names (i.e. `delay`) more understandable in `NetworkStatusTimeline.cs`.
4. Address comments and obtain 5-10 passing runs for single node connectivity and it is safe to merge.

File: test/modules/TestResultCoordinator/Topology.cs [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-    public enum Topology
-    {
-        // A single edge device.
-        SingleNode,
-
-        // Multiple edge devices connected together in some toplogy.
-        // Runs an extended test suite relative to `SingleNode`.
-        Nested
-    }
-}
```
Comment (and-rewsmith) [inline test/modules/TestResultCoordinator/Topology.cs:13]: What was the change to this file? We may want to revert the change.

File: test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodLongHaulReportGeneratorTest.cs [hunk 6]
```diff
@@ -11,6 +11,7 @@ namespace Modules.Test.TestResultCoordinator.Reports.DirectMethod
     using Microsoft.Azure.Devices.Edge.ModuleUtil;
     using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
     using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
     using Moq;
     using Newtonsoft.Json;
     using Xunit;
```
Comment (and-rewsmith) [inline test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodLongHaulReportGeneratorTest.cs:17]: May want to revert changes to this file.

**Review by and-rewsmith — DISMISSED** [review]
A few nit comments but if this has multiple successive passes I think it is good to go. If we encounter any failures we will consider adding a tolerance and following up with service side investigation.

**Review by huguesBouvier — APPROVED** [review]

File: test/modules/TestResultCoordinator/Reports/DirectMethod/LongHaul/DirectMethodLongHaulReport.cs [hunk 7]
```diff
@@ -51,7 +51,7 @@ class DirectMethodLongHaulReport : TestResultReportBase
 
         public override bool IsPassed => this.IsPassedHelper();
 
-        bool IsPassedHelper()
+        private bool IsPassedHelper()
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/Reports/DirectMethod/LongHaul/DirectMethodLongHaulReport.cs:54]: Remove

File: e2e_deployment_files/connectivity_deployment.template.json (hunk 2 above)
Comment (brotherneeru) [reply e2e_deployment_files/connectivity_deployment.template.json:900]: From my understanding , When Hugues originally did testing on this, he noticed that there seemed to be some arbitrary tolerance period implemented for when the directmethod test messages started coming in. It had to be 1 minute after network controller was created. Doing some investigating, some of the messages started coming in after 1 minute and the tests were considering this an error. Increasing the tolerance period to 3 minutes would catch these

File: test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodLongHaulReportGeneratorTest.cs [hunk 8]
```diff
@@ -11,6 +11,7 @@ namespace Modules.Test.TestResultCoordinator.Reports.DirectMethod
     using Microsoft.Azure.Devices.Edge.ModuleUtil;
     using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
     using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
```
[the remark this answers is not in the document]
Comment (varunpuranik) [reply test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodLongHaulReportGeneratorTest.cs:14]: Can you please revert this change?

File: test/modules/TestResultCoordinator/Reports/DirectMethod/LongHaul/DirectMethodLongHaulReport.cs (hunk 7 above)
Comment (varunpuranik) [reply test/modules/TestResultCoordinator/Reports/DirectMethod/LongHaul/DirectMethodLongHaulReport.cs:54]: Can you please address this?

File: test/modules/TestResultCoordinator/Topology.cs (hunk 3 above)
Comment (varunpuranik) [reply test/modules/TestResultCoordinator/Topology.cs:1]: Can you please address this?

Comment (brotherneeru) [reply test/modules/TestResultCoordinator/Topology.cs:1]: All I did here was remove the extra newline at the bottom of the file initially

File: test/modules/TestResultCoordinator/Topology.cs (hunk 5 above)
Comment (brotherneeru) [reply test/modules/TestResultCoordinator/Topology.cs:13]: When I originally pulled in these, it had an extra newline at the bottom of the file initially

File: test/modules/TestResultCoordinator/Topology.cs (hunk 3 above)
Comment (and-rewsmith) [reply test/modules/TestResultCoordinator/Topology.cs:1]: I think when your editor saved it might have added some line endings or maybe something else. You can use `git checkout <upstream_branch> -- test/modules/TestResultCoordinator/Topology.cs` to revert.

**Review by varunpuranik — DISMISSED** [review]

**Review by and-rewsmith — APPROVED** [review]

**Review by varunpuranik — APPROVED** [review]
